core/app/models/spree/gallery/variant_gallery.rb

```diff
@@ -0,0 +1,38 @@
+module Spree
+  module Gallery
+    class VariantGallery
+      def initialize(variant)
+        @variant = variant
+      end
+
+      # A list of all images associated with this gallery
+      #
+      # @return [Enumerable<Spree::Image>] all images in the gallery
+      def images
+        @variant.images
+      end
+
+      # The "primary" (first) {Spree::Image} associated with the gallery
+      #
+      # @return [Spree::Image] primary image for the gallery
+      def primary_image
+        @variant.images.first || Spree::Image.new
+      end
+
+      # The "best" {Spree::Image} associated with the gallery
+      # This will return the first associated product image if there are no
+      # variant images
+      #
+      # @return [Spree::Image] best image for the variant gallery
+      def best_image
+        @variant.images.first || product_images.first || Spree::Image.new
+      end
+
+      private
+
+      def product_images
+        @variant.product.variant_images
```

Comment:
I don’t see a need for this private method as long as we don’t access it from two different methods at least.